### DISCUSSION TITLE
Exploring possibility of using analog for docs

### DISCUSSION THREAD

**Discussion by santoshyadavdev** [analogjs/analog#779] [Ideas]
Currently we are using docusaurus for documentation, but analog has the capability for rendering markdown, how about exploring the possibility of using analog for documentation, and maybe in future we can provide an doc template similar to what astro did with starlight.

**Comment by brandonroberts** [comment]
I think this is doable. @goetzrobin is using Analog for the https://spartan.ng docs. Needs more pieces to use out of the box as a documentation platform.

**Reply by goetzrobin** [reply]
Currently, I am using it in a pretty hands on actual Angular type of way. Each page is an Angular component and the documentation is written in the template instead of beautiful easy markdown...

Do you guys think we would need mdx support for this? I could see us doing something similar to the `contentPlugin` to read and parse the `docs` folder in the documentation app. I am wondering what an MVP would need to be able to do...

**Comment by erkamyaman** [comment]
> Currently we are using docusaurus for documentation, but analog has the capability for rendering markdown, how about exploring the possibility of using analog for documentation, and maybe in future we can provide an doc template similar to what astro did with starlight.

I've been building NgMd to scratch this itch. It's an Angular docs starter on AnalogJS, Vite, Tailwind, and Shiki. You drop a markdown file under src/content/ and get a routed page with a sidebar, search, link guards, and a sitemap.

It's at v0.1.7 right now, but it has a lot of potential to grow. The roadmap to 1.0 covers API reference generation, versioning, and i18n.

Scaffold: `pnpm create ngmd@latest my-docs`
Repo: https://github.com/erkamyaman/ngmd
Live: https://ngmd.netlify.app

Thanks to @santoshyadavdev for raising the question in the first place, and to @brandonroberts for building AnalogJS, which NgMd is built on.
